src/sortStudents.ts

```diff
@@ -1,16 +1,128 @@
 
 export interface Student {
-  // describe Student interface
+  name: string,
+  surname: string,
+  age: number,
+  married: boolean,
+  grades: number[],
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name = 'name',
+  Surname = 'surname',
+  Age = 'age',
+  Married = 'married',
+  AverageGrade = 'grades',
 }
 
-// create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
 
+export function sortStudents(
+  students: Student[],
+  sortBy: SortType,
+  order: SortOrder,
+): Student[] {
+  const resArr: Student[] = [...students];
+  const sortOrder: boolean = order === 'asc';
 
-export function sortStudents(students, sortBy, order) {
-  // write your function
+  switch (sortBy) {
+    case SortType.Name:
+      resArr.sort((a: Student, b: Student): number => {
+        if (sortOrder) {
+          return a.name.localeCompare(b.name);
+        }
+
+        return b.name.localeCompare(a.name);
+      });
+      break;
+    case SortType.Surname:
+      resArr.sort((a: Student, b: Student): number => {
+        if (sortOrder) {
+          return a.surname.localeCompare(b.surname);
+        }
+
+        return b.surname.localeCompare(a.surname);
+      });
+      break;
+    case SortType.Age:
+      resArr.sort((a: Student, b: Student): number => {
+        let res = 0;
+
+        if (Number(a.age) > Number(b.age)) {
+          if (sortOrder) {
+            res = 1;
+          } else {
+            res = -1;
+          }
+        }
+
+        if (Number(a.age) < Number(b.age)) {
+          if (!sortOrder) {
+            res = 1;
+          } else {
+            res = -1;
+          }
+        }
+
+        return res;
```

Comment:
you already have numbers here, don't need to use `Number`.

You could simply use the ternary operator, without `res`. Simplify this way all over the code.
```suggestion
   return order === 'asc' 
   ? a.age - b.age
   : b.age - a.age;
```